plugins/reolink/src/main.ts

```diff
@@ -295,6 +316,14 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
             && this.storageSettings.values.abilities?.value?.Ability?.supportAudioAlarm?.ver !== 0;
     }
 
+    hasFloodlight() {
+        const channel = this.getRtspChannel();
+        const floodLightConfigVer = this.storageSettings.values.abilities?.value?.Ability?.abilityChn?.[channel].floodLight?.ver ?? 0;
```

Comment:
optional chain the index. use the same variable so you're not evaluating the same thing 3 times.

Reply:
done